lib/flipper/middleware/eager_memoizer.rb

```diff
@@ -0,0 +1,48 @@
+require 'rack/body_proxy'
+
+module Flipper
+  module Middleware
+    class EagerMemoizer
```

Comment:
What do you think about calling this Flipper::Middleware::PreloadAll? 

Also, what do you think about having the preload all memoizer only call `preload_all`. We could then recommend including the memoizer first in the stack and `PreloadAll` second. It would keep logic in the same place. 

In fact, I think we could even set an env var like flipper.memoizing in the memoizing middleware and then log a warning to stderr or something here if it is missing, as a way of letting people know they are preloading without memoization. 

Another option, we could just add a preload_all option to the memoizing middleware so people could just do:

```ruby
use Flipper::Middleware::Memoizer, preload_all => true

# or 
use Flipper::Middleware::Memoizer, preload: [:my, :collection, :of, :features, :to, :preload]
```